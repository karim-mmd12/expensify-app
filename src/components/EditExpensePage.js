import React from 'react'
import ExpenseForm from './ExpenseForm'
import { connect } from 'react-redux'
import { editExpense, startRemoveExpense } from '../actions/expenses'

const EditExpensePage = (props) => {
  console.log(props)
  return (
    <div>
      <ExpenseForm
        expense={props.expense}
        onSubmit={(expense) => {
          props.dispatch(editExpense(props.expense.id, expense))
          props.history.push('/')
        }}
      />

      <button
        onClick={() => {
          console.log(props.expense)
          props.dispatch(startRemoveExpense({ id: props.expense.id }))
          props.history.push('/')
        }}
      >
        Remove
      </button>
    </div>
  )
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id),
  }
}
export default connect(mapStateToProps)(EditExpensePage)
