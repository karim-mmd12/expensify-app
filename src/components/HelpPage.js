import React from 'react';

const HelpPage = () => (
    <div>
        This is from HelpPage Component.
    </div>
);
export default HelpPage;