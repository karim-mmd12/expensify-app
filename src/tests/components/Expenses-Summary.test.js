import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/Expenses-Summary';

test('should render expenses summary component correctly with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={32.45} />);
    expect(wrapper).toMatchSnapshot();
});
test('should render expenses summary component correctly with many expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={3} expensesTotal={104774552.45} />);
    expect(wrapper).toMatchSnapshot();
});