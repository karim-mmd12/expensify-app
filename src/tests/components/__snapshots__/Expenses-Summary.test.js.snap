// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenses summary component correctly with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $0.32
  </h1>
</div>
`;

exports[`should render expenses summary component correctly with many expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $1,047,745.52
  </h1>
</div>
`;
